src/pcrlock/pcrlock.c

```diff
@@ -4416,6 +4416,9 @@ static int make_policy(bool force, RecoveryPinMode recovery_pin_mode) {
         if (r < 0)
                 return r;
 
+        if (!new_prediction.pcrs)
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "Set of PCRs to use for policy is empty");
```

Comment:
hmm, i think it might make sense to print a log message at LOG_NOTICE level or so when this happens, but I don't think we should fail in this case. It's quite a common case that during install time we cannot recognize anything in the PCRs properly (because we never ran before properly), but that shouldn't stop us from already setting up the nvindex with this so that we can enroll a disk with it. (after all, the enrollment strictly pings the nvindex on the tpm, hence we want to do this asap even if we have no policy to place in it yet).

later, during the subsequent boot things might change and we can then write a more useful policy into the already set up nvindex without having to bother we re-enrollments, they'll just work.

Reply:
While I understand the use case, I'm not sure always allowing this is a good idea. If all PCRs are excluded or are having the empty pcrlock.d directory issue, the user will call make-policy, which generate a policy with zero digest and unlock may happen whatever is configured. I've not tested on real HW but with qemu:

```
# mkdir -p /var/lib/pcrlock.d/700-empty.d
# systemd-pcrlock make-policy --pcr=4,7
...
PCRs in protection mask: 4 (boot-loader-code), 7 (secure-boot-policy)
Predicted future PCRs in 118us.
[]
...
Written policy digest 0000000000000000000000000000000000000000000000000000000000000000 to NV index 0x1b9a44e
...
# tpm2_pcrread sha256:4,7

  sha256:
    4 : 0x0E06CD8E7244D9BA9229062E9EC5337F3753F595A75DE6B0033AE1AA76365433
    7 : 0x0D82B7DA81B8FC0CEDFC0F6A2D7C6A58C39B1FA20BD9AFE4885EEAD3D0CACC30
```

Reboot with a different bootloader for instance and it's unlocking
```
# tpm2_pcrread sha256:4,7
  sha256:
    4 : 0xDE28954A1C8EFFD79BA527D056452214CE9617BF0EC287AAFA1AD291F1623CE6
    7 : 0x254520B8B4E9339DF14184DCF471EAD247178C61F383DEDF59E3FD65D46281AC
```

I'm not sure that it's a good thing.

Reply:
that's entirely fine I think. Sure deserves a message being logged, but is not really a problem. If people want to define policies like that it's entirely ok to do so, and as mentioned that's not just theoretic, but very practical too

Reply:
also, this should be an explicit numeric comparison i.e. `!= 0`

Reply:
if the user is wanting that, it may be fine (even if a policy validating everything is no protection imho). What I'm worried is cases where the user didn't want that and is believing that it's protected.

What about:
- return an error by default
- add a new command line option to explicitly tell systemd-pcrlock that it's fine to lock on an empty PCR list ?

Reply:
Sorry, but no, failing in this case is not acceptable. It's a common case that if we have no previous PCR measurements we install a zero policy (and thus allocate the nvindex, we can already reference). 

Yes, a log message highlighting this is OK, but it should be LOG_NOTICE at best.